com.github.opentyrian.OpenTyrian.yaml

```
@@ -0,0 +1,58 @@
+app-id: com.github.opentyrian.OpenTyrian
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+finish-args:
+  - --device=all
+  - --share=network
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  # Pulseaudio access
+  - --socket=pulseaudio
+  # OpenGL access
+  - --device=dri
+  #SDL needs to be able to talk to UPower
+  - --system-talk-name=org.freedesktop.UPower
+  #Allow SDL to inhibit the screensaver
+  - --talk-name=org.freedesktop.ScreenSaver
+command: opentyrian_start
+rename-appdata-file: opentyrian.appdata.xml
+rename-desktop-file: opentyrian.desktop
+rename-icon: opentyrian
+copy-icon: true
+modules:
+  - name: opentyrian
+    buildsystem: simple
+    build-commands:
+      - make install DESTDIR=${FLATPAK_DEST}/ prefix=
+    post-install:
+      - install -Dm755 apply_extra ${FLATPAK_DEST}/bin/apply_extra
+      - install -Dm755 opentyrian_start ${FLATPAK_DEST}/bin/opentyrian_start
+      - install -Dm644 opentyrian.appdata.xml -t ${FLATPAK_DEST}/share/metainfo/
+      - install -Dm644 linux/opentyrian.desktop -t ${FLATPAK_DEST}/share/applications/
+      - desktop-file-edit --set-key=Exec --set-value=opentyrian_start ${FLATPAK_DEST}/share/applications/opentyrian.desktop
+      - install -Dm644 linux/icons/tyrian-128.png ${FLATPAK_DEST}/share/icons/hicolor/128x128/apps/opentyrian.png
+      - install -Dm644 linux/icons/tyrian-48.png ${FLATPAK_DEST}/share/icons/hicolor/48x48/apps/opentyrian.png
+      - install -Dm644 linux/icons/tyrian-32.png ${FLATPAK_DEST}/share/icons/hicolor/32x32/apps/opentyrian.png
+      - install -Dm644 linux/icons/tyrian-24.png ${FLATPAK_DEST}/share/icons/hicolor/24x24/apps/opentyrian.png
+      - install -Dm644 linux/icons/tyrian-22.png ${FLATPAK_DEST}/share/icons/hicolor/22x22/apps/opentyrian.png
+    sources:
+      - type: git
+        url: https://github.com/opentyrian/opentyrian
+        tag: v2.1.20220318
+        commit: 8f205e6b12cb9c34905fa960416b6f661bf38ac3
+      - type: file
+        path: opentyrian.appdata.xml
+      - type: file
+        path: opentyrian_start
+      - type: extra-data
+        filename: tyrian21.zip
+        url: https://camanis.net/tyrian/tyrian21.zip
+        sha256: 7790d09a2a3addcd33c66ef063d5900eb81cc9c342f4807eb8356364dd1d9277
+        size: 4754048
+      - type: script
+        dest-filename: apply_extra
+        commands:
+          - unzip -q tyrian21.zip
+          - rm tyrian21.zip
```

```suggestion
          - rm tyrian21.zip
          - mkdir -p ${FLATPAK_DEST}/share/tyrian
          - mv ${FLATPAK_DEST}/extra/tyrian21 ${FLATPAK_DEST}/share/tyrian
```

---

`mkdir: cannot create directory ‘/usr/local’: Read-only file system`

How to solve it?

---

Ah, yes that is confusing. The prefix changes at runtime.

Try
```suggestion
          - rm tyrian21.zip
          - mv tyrian21 tyrian
```

---

I don't think `FLATPAK_DEST` is set when running `apply_extra`.

---

> @Mailaender
> 
> See current build:
> 
> `mkdir: cannot create directory ‘/usr/local’: Read-only file system`
> 
> How to solve it?

see my comment above for the build command.